Content.Server/Light/EntitySystems/ToggleableOccluderSystem.cs

```diff
@@ -0,0 +1,58 @@
+using Content.Server.DeviceLinking.Events;
+using Content.Server.DeviceLinking.Systems;
+using Content.Server.Light.Components;
+using Robust.Shared.GameObjects;
+
+namespace Content.Server.Light.EntitySystems
+{
+    /// <summary>
+    ///     Hadnles the logic between signals and toggling OccluderComponent
+    /// </summary>
+    public sealed class ToggleableOccluderSystem : EntitySystem
+    {
+        [Dependency] private readonly DeviceLinkSystem _signalSystem = default!;
+        [Dependency] private readonly OccluderSystem _occluder = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ToggleableOccluderComponent,SignalReceivedEvent>(OnSignalReceived);
+            SubscribeLocalEvent<ToggleableOccluderComponent,ComponentInit>(OnInit);
+        }
+
+        private void OnInit(EntityUid uid, ToggleableOccluderComponent comp, ComponentInit args)
+        {
+            if(TryComp<OccluderComponent>(uid, out var occluder))
+                _signalSystem.EnsureSinkPorts(uid, comp.OnPort, comp.OffPort, comp.TogglePort);
+        }
+
+        private void OnSignalReceived(EntityUid uid, ToggleableOccluderComponent comp, ref SignalReceivedEvent args)
+        {
+            TryComp<OccluderComponent>(uid, out var occluder);
```

Comment:
```suggestion
            if (!TryComp<OccluderComponent>(uid, out var occluder))
                return;
```
In case OccluderComponent does not exist